MHV-55122: Update landing and detail page for multiselect

vmenshutin-bylight opened the pull request:
## Summary

- Updated Landing Page (a.k.a. About Medications page) and Medication Details page content
- Hid the content behind the existing feature toggle

## Related issue(s)

https://jira.devops.va.gov/browse/MHV-55122

<img width="933" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/133991282/811472f0-4409-41e7-9c2c-03ae4a9c216f">

## Testing done

- Manual testing

## What areas of the site does it impact?

/my-health/medications/

## Acceptance criteria

AC1: New content for the header on the landing page, remove the informational body text
AC2: Add action link to begin the new pages/flow
AC3: Update content, changed colors to blue
AC4: For fillable/refillable meds on the details page the button should change from a button to a primary green action link that takes them to new refill page.

Sketch Link:[Medications – Figma](https://www.figma.com/file/UGHasFKuKENvKz8AgSmxtI/Medications?type=design&node-id=79-3897&mode=design)

## Screenshots

<img width="929" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/133991282/2e436281-b919-478d-9f99-d257fdcfdfc6">

<img width="601" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/133991282/a8004d3d-af19-4f56-845b-66a4ed3b486a">

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

vmenshutin-bylight committed:
MHV-55122: Update landing and detail page for multiselect

agravell047 commented on src/applications/mhv/medications/components/PrescriptionDetails/VaPrescription.jsx:
```diff
@@ -30,7 +37,17 @@ const VaPrescription = prescription => {
+              <a
+                className="vads-u-display--block vads-c-action-link--green vads-u-margin-top--3 vads-u-margin-bottom--3"
+                href={medicationsUrls.MEDICATIONS_REFILL}
+                data-testid="refill-nav-link"
```
Should we add a test for this new data-testid?

Also just curious, was there a specific reason to use `<a>` instead of `<Link>` here?

vmenshutin-bylight replied:
changing <a> to <Link> now. as for the unit test - I do not see how would it be of a value: the link is not dynamic and such functionality is usually covered by e2e (+ unit test snapshot). if you were referring to feature toggle condition - we are planning to clean it anyway

thank you for your comment btw!!!

vmenshutin-bylight committed:
Adressed PR comments

vmenshutin-bylight committed:
minor cleaning

Commit:
verify refill link on landing and details page

added tests to verify refill prescriptions link on landing page and details page
